operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -0,0 +1,231 @@
+use core::slice;
+use futures::StreamExt;
+use gdal::{
+    raster::{Buffer, GdalType},
+    Driver,
+};
+use gdal_sys::{VSIFree, VSIGetMemFileBuffer};
+use geoengine_datatypes::{
+    primitives::{SpatialBounded, TimeInterval},
+    raster::{Blit, GeoTransform, Grid2D, GridShape2D, GridSize, Pixel, RasterTile2D},
+    spatial_reference::SpatialReference,
+};
+use std::{
+    convert::TryInto,
+    sync::mpsc::{Receiver, Sender},
+};
+use std::{ffi::CString, sync::mpsc};
+
+use crate::util::Result;
+use crate::{
+    engine::{QueryContext, QueryRectangle, RasterQueryProcessor},
+    error::Error,
+};
+
+pub async fn raster_stream_to_geotiff_bytes<T, C: QueryContext + 'static>(
+    processor: Box<dyn RasterQueryProcessor<RasterType = T>>,
+    query_rect: QueryRectangle,
+    query_ctx: C,
+    width: u32,
+    height: u32,
+    no_data_value: Option<f64>,
+    spatial_reference: SpatialReference,
+) -> Result<Vec<u8>>
+where
+    T: Pixel + GdalType,
+{
+    let file_name = format!("/vsimem/{}.tiff", uuid::Uuid::new_v4());
+    let (tx, rx): (Sender<RasterTile2D<T>>, Receiver<RasterTile2D<T>>) = mpsc::channel();
+
+    let file_name_clone = file_name.clone();
+    let writer = tokio::task::spawn_blocking(move || {
+        gdal_writer(
+            &rx,
+            &file_name_clone,
+            query_rect,
+            width,
+            height,
+            no_data_value,
+            spatial_reference,
+        )
+    });
+
+    let mut tile_stream = processor.raster_query(query_rect, &query_ctx).await?;
+
+    while let Some(tile) = tile_stream.next().await {
+        tx.send(tile?).map_err(|_| Error::ChannelSend)?;
+    }
+
+    drop(tx);
+
+    writer.await??;
+
+    // TODO: use higher level rust-gdal method when it is mapped
+    let bytes = get_vsi_mem_file_bytes_and_free(&file_name);
+
+    Ok(bytes)
+}
+
+fn gdal_writer<T: Pixel + GdalType>(
+    rx: &Receiver<RasterTile2D<T>>,
+    file_name: &str,
+    query_rect: QueryRectangle,
+    width: u32,
+    height: u32,
```

Comment:
why do you need width and hight? the query rectangle has a resolution

Reply:
9a47142823e2749a0e7a7c730b4de82ff969cda7

Reply:
great but i think it is not a good idea to hardcode the tile size 600 which was selected randomly

Reply:
this is just a test